Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp

```diff
@@ -209,7 +210,7 @@ namespace AZ::Reflection
                 {
                     return;
                 }
-                EditContext::EnumerateInstanceCallContext context(
+            SerializeContext::EnumerateInstanceCallContext context(
```

Comment:
Yeah, the `EditContext::EnumerateInstanceCallContext context` doesn't reflect the edit context from any of the base classes, so upon recommendation from @lumberyard-employee-dm we changed it to use the serialize context enumerate.

If it's possible to change the edit context enumerate to also reflect the base classes, that would be a good fix that I can try

Reply:
The Edit context has no relation of base to derived classes. 

There are several ways to add base class information to the Edit Context.
One is to have the user explicitly specify the base class element in reflection of the EditContext itself, another way is to implicitly reflect an EditContext `ElementData` isntance using the SerializeContext to retrieve the base class data.

- Add support for the user to explicitly specify a base class element through a new EditContext function. Perhaps through a `BaseElement` function on the ClassBuilder that adds a `Edit::ElementData` for a base type.

### Benefits
* It is easy to attach attributes to the Base class data elements. This allows easier control the visibility of the base class user has explicit control

### Problems
*Requires changes to EditContext reflections to replicate the ReflectedPropertyEditor visibility. This could involve adding 100s of `BaseElement` function calls across the codebase to replicate the visibility

* Implicitly add the `Edit::ElementData` elements for each reflected base type associated with the reflected type in SerializeContext.

### Benefits
* The user doesn't have to add the base class reflection to the Edit Context explicitly

### Problems
* The [EditContextConstants.inl](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Serialization/EditContextConstants.inl#L17-L40) would be needed for attaching attributes to the Base Class elements though (Perhaps attributes like "BaseClass1",  "BaseClass2", ... , "BaseClassN").
The EditContext would need code updates to use the new BaseClass attributes to determine which base class elements are to attach other attributes into.
i.e
```c++
 editContext->Class<TestContainer>("TestContainer", "")
    ->ClassElement(AZ::Edit::ClassElements::BaseClass1, "BaseClass")
    ->Attribute(AZ::Edit::Attributes::Visibility, PropertyVisibility::ShowChildrenOnly)
   ;
```